### PR TITLE
added mll bins, cleaned up W and Z (models, generate syntax, cuts, mi…

### PR DESCRIPTION
**Pull request by agrohsje**
…ssing madspin file for W)

### CODE REVIEW

File: Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8.json~ [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (sihyunjeon) [inline Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8.json~:1]: Is this file needed? It has ~ in the back

File: Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8_cuts.f [hunk 2]
```diff
@@ -1,4 +1,4 @@
-c *This file contains cuts(10~50) on the mll: line142-148*
+c ADDED: veto on invariant masses above 50 GeV in line 142-148
```
Comment (sihyunjeon) [inline Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8_cuts.f:1]: Should we ask Sanghyun or someone if this is mg29x useable cuts.f file? I think at some point it changed but not sure in which version

File: Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-50_2J_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-50_2J_amcatnloFXFX-pythia8.json~ [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (sihyunjeon) [inline Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-50_2J_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-50_2J_amcatnloFXFX-pythia8.json~:1]: Also json~

File: Cards/MadGraph5_aMCatNLO/DY/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (sihyunjeon) [inline Cards/MadGraph5_aMCatNLO/DY/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8.json:1]: A bit wide scope question, not about the card itself. Do we really need DY MLM 10to50? I am wondering who might be the users for this. Maybe some SUS with lepton jet multiple final states?

Comment (sihyunjeon) [reply Cards/MadGraph5_aMCatNLO/DY/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8.json:1]: Otherwise, cards look fine

File: Cards/MadGraph5_aMCatNLO/DY/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8_proc_card.dat [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+define ell+ = e+ mu+ ta+
+define ell- = e- mu- ta-
+define vl = ve vm vt
+define vl~ = ve~ vm~ vt~
+
+generate p p > ell+ ell-  @0
+add process p p > ell+ ell- j  @1
+add process p p > ell+ ell- j j  @2
+add process p p > ell+ ell- j j j  @3
+add process p p > ell+ ell- j j j j  @4
```
Comment (sihyunjeon) [inline Cards/MadGraph5_aMCatNLO/DY/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8_proc_card.dat:12]: Double checking, we don't have H contributions from taus right?

File: Cards/MadGraph5_aMCatNLO/W/WtoLNu-2Jets_0J_amcatnloFXFX-pythia8/WtoLNu-2Jets_0J_amcatnloFXFX-pythia8_madspin_card.dat [hunk 6]
```diff
@@ -0,0 +1,9 @@
+set ms_dir ./madspingrid
+
+set max_running_process 1
+set onlyhelicity True
+
+decay w+ > ta+ vt
+decay w- > ta- vt~
```
Comment (sihyunjeon) [inline Cards/MadGraph5_aMCatNLO/W/WtoLNu-2Jets_0J_amcatnloFXFX-pythia8/WtoLNu-2Jets_0J_amcatnloFXFX-pythia8_madspin_card.dat:7]: I think these "decay XXX" lines were no longer needed when we do "set onlyhelicity True"

File: Cards/MadGraph5_aMCatNLO/W/WtoLNu-2Jets_1J_amcatnloFXFX-pythia8/WtoLNu-2Jets_1J_amcatnloFXFX-pythia8_madspin_card.dat [hunk 7]
```diff
@@ -0,0 +1,9 @@
+set ms_dir ./madspingrid
+
+set max_running_process 1
+set onlyhelicity True
+
+decay w+ > ta+ vt
+decay w- > ta- vt~
```
Comment (sihyunjeon) [inline Cards/MadGraph5_aMCatNLO/W/WtoLNu-2Jets_1J_amcatnloFXFX-pythia8/WtoLNu-2Jets_1J_amcatnloFXFX-pythia8_madspin_card.dat:7]: Same comment - double check with Jie or Sanghyun

File: Cards/MadGraph5_aMCatNLO/W/WtoLNu-2Jets_2J_amcatnloFXFX-pythia8/WtoLNu-2Jets_2J_amcatnloFXFX-pythia8_madspin_card.dat [hunk 8]
```diff
@@ -0,0 +1,9 @@
+set ms_dir ./madspingrid
+
+set max_running_process 1
+set onlyhelicity True
+
+decay w+ > ta+ vt
+decay w- > ta- vt~
```
Comment (sihyunjeon) [inline Cards/MadGraph5_aMCatNLO/W/WtoLNu-2Jets_2J_amcatnloFXFX-pythia8/WtoLNu-2Jets_2J_amcatnloFXFX-pythia8_madspin_card.dat:7]: Same

File: Cards/MadGraph5_aMCatNLO/W/WtoLNu-2Jets_amcatnloFXFX-pythia8/WtoLNu-2Jets_amcatnloFXFX-pythia8_madspin_card.dat [hunk 9]
```diff
@@ -0,0 +1,9 @@
+set ms_dir ./madspingrid
+
+set max_running_process 1
+set onlyhelicity True
+
+decay w+ > ta+ vt
+decay w- > ta- vt~
```
Comment (sihyunjeon) [inline Cards/MadGraph5_aMCatNLO/W/WtoLNu-2Jets_amcatnloFXFX-pythia8/WtoLNu-2Jets_amcatnloFXFX-pythia8_madspin_card.dat:7]: Same

File: Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8_cuts.f (hunk 2 above)
Comment (agrohsje) [reply Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8_cuts.f:1]: When I made the file, I checked out 299 and diffed cuts.f. The files are identical except the added lines.

File: Cards/MadGraph5_aMCatNLO/DY/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8.json (hunk 4 above)
Comment (agrohsje) [reply Cards/MadGraph5_aMCatNLO/DY/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8.json:1]: Right. Usually the stats from this is super low.

File: Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8.json~ (hunk 1 above)
Comment (agrohsje) [reply Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8.json~]: my mistake

File: Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-50_2J_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-50_2J_amcatnloFXFX-pythia8.json~ (hunk 3 above)
Comment (agrohsje) [reply Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-50_2J_amcatnloFXFX-pythia8/DYto2L-2Jets_MLL-50_2J_amcatnloFXFX-pythia8.json~]: sorry. missed that one as well. removed.

File: Cards/MadGraph5_aMCatNLO/DY/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8_proc_card.dat (hunk 5 above)
Comment (agrohsje) [reply Cards/MadGraph5_aMCatNLO/DY/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8/DYto2L-4Jets_MLL-10to50_madgraphMLM-pythia8_proc_card.dat:12]: I tried but didn't spot any difference in diagrams.

File: Cards/MadGraph5_aMCatNLO/W/WtoLNu-2Jets_0J_amcatnloFXFX-pythia8/WtoLNu-2Jets_0J_amcatnloFXFX-pythia8_madspin_card.dat (hunk 6 above)
Comment (agrohsje) [reply Cards/MadGraph5_aMCatNLO/W/WtoLNu-2Jets_0J_amcatnloFXFX-pythia8/WtoLNu-2Jets_0J_amcatnloFXFX-pythia8_madspin_card.dat]: Removed. Thanks for pointing out!

**Comment by agrohsje** [comment]
Hi @sihyunjeon. Do you mind merging?

**Comment by sihyunjeon** [comment]
Will you trigger the submission?

**Comment by agrohsje** [comment]
Yes. Will run tests with 1 jet for the 3 NLO setups.
